### PR TITLE
autofocus properly during the reg flow

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @cvializ I'm guessing there is an easier way to do this

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/signup/register.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+angular.module('kifi')
+
+.directive('kfRegister', [
+  function() {
+    return {
+      restrict: 'A',
+      require: '^kfModal',
+      $scope: {},
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/signup/register.js:10]: the dollar sign isn't needed here

File: kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    <div class="dialog-body">
+      <div ng-if="userData.invite">
+        <div class="kf-register-header kf-register-header-centered">Join Kifi</div>
+        <div>
+          <img class="kf-register-inviter-pic" ng-if="userData.invite" ng-src="{{userData.invite.inviter|pic}}"></img>
+            <span class="kf-register-modal-invite-text" ng-if="userData.invite && userData.intent === 'follow'">
+              {{userData.invite.inviter.firstName}} invited you to {{userData.invite.access === 'read_only' ? 'follow' : 'collaborate on'}} this library.
+            </span>
+            <span class="kf-register-modal-invite-text" ng-if="userData.invite && userData.intent === 'joinOrg'">
+              {{userData.invite.inviter.firstName}} invited you to become a member of this team.
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html:16]: Everywhere you used `{{ blah }}` replace it with `ng-bind="blah"`, like here you would use `<span ng-bind="userData.invite..."></span>`

File: kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+         --><span ng-if="fieldHasError(regEmailForm.email)" class="register-help-block">Enter a valid email address</span>
+          </span><!--
+       --><input type="submit" value="Continue" class="register-submit register-continue">
+    </div>
+
+    <div class="register-tagline-login">Already a Kifi member?&nbsp; <a ng-href="/login?{{authTokenQueryParam}}" target="_self" ng-click="onLoginClick()">Log in</a></div>
+  </div>
+  <script>
+    $(".register-email-box").focus();
+  </script>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html:55]: Whoa nope. If you want some code to run after the template is rendered use `$scope.$evalAsync(function () {})` (and if that doesn't work use `$timeout(function () {})`. the execution order can be strange) and put it in the directive's `link` function

File: kifi-backend/modules/shoebox/angular/src/signup/registerFinalize.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+angular.module('kifi')
+
+.directive('kfRegisterFinalize', [
+  function() {
+    return {
+      restrict: 'A',
+      require: '^kfModal',
+      $scope: {},
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/signup/registerFinalize.js:10]: no dollar sign

**Comment by cvializ** [comment]
nice work!

**Comment by camhashemi** [comment]
thank you @cvializ. The end goal is to have auto-focus work properly, but it seemed like with the dynamic modals popping up this was the best way for me to assert control. The JQuery script was left over from one of my attempts, since I didn't really want to make two directives just for autofocus. I was thinking that some code in `modalService` (the closest parent directive before these two new ones) could do it, but this worked and could make these modals easier to work with in the future.

File: kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html (hunk 2 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html:16]: oh I just noticed this is copypasted from below

Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/signup/register.tpl.html:16]: that's fine, still fixed it.

**Comment by cvializ** [comment]
maybe one directive without a template could do it, like some kind of `kf-autofocus`, with the `$scope.$evalAsync` (or `$timeout`, I can't remember if the former works, but I prefer it if it does)

**Comment by camhashemi** [comment]
we have a kf-autofocus... :o

**Comment by cvializ** [comment]
: O

**Pull request closed without merging**